validator/accounts/v2/accounts_delete.go

```diff
@@ -29,47 +29,44 @@ func DeleteAccount(cliCtx *cli.Context) error {
 	if err != nil {
 		return errors.Wrap(err, "could not initialize keymanager")
 	}
-	allAccounts, err := keymanager.FetchValidatingPublicKeys(ctx)
+	validatingPublicKeys, err := keymanager.FetchValidatingPublicKeys(ctx)
 	if err != nil {
 		return err
 	}
-	if len(allAccounts) == 0 {
+	if len(validatingPublicKeys) == 0 {
 		return errors.New("wallet is empty, no accounts to delete")
 	}
-	allAccountStrs := make([]string, len(allAccounts))
-	for i, account := range allAccounts {
-		allAccountStrs[i] = fmt.Sprintf("%#x", bytesutil.FromBytes48(account))
-	}
-	accounts, err := selectAccounts(cliCtx, selectAccountsDeletePromptText, allAccountStrs)
+	// Allow the user to interactively select the accounts to backup or optionally
+	// provide them via cli flags as a string of comma-separated, hex strings.
+	filteredPubKeys, err := filterPublicKeysFromUserInput(
+		cliCtx,
+		flags.DeletePublicKeysFlag,
+		validatingPublicKeys,
+		selectAccountsDeletePromptText,
+	)
 	if err != nil {
-		return err
-	}
-	if len(accounts) == 0 {
-		return errors.New("no accounts selected to delete")
+		return errors.Wrap(err, "could not filter public keys for deletion")
 	}
-
-	formattedPubKeys := make([]string, len(accounts))
-	for i, pubKey := range accounts {
-		keyBytes, err := hex.DecodeString(pubKey[2:])
-		if err != nil {
-			return errors.Wrap(err, "could not decode hex string")
-		}
-		formattedPubKeys[i] = fmt.Sprintf("%#x", bytesutil.Trunc(keyBytes))
+	rawPublicKeys := make([][]byte, len(filteredPubKeys))
+	formattedPubKeys := make([]string, len(filteredPubKeys))
+	for i, pk := range filteredPubKeys {
+		pubKeyBytes := pk.Marshal()
+		rawPublicKeys[i] = pubKeyBytes
+		formattedPubKeys[i] = fmt.Sprintf("%#x", bytesutil.Trunc(pubKeyBytes))
 	}
 	allAccountStr := strings.Join(formattedPubKeys, ", ")
-
-	if len(accounts) == 1 {
+	if len(filteredPubKeys) == 1 {
 		promptText := "Are you sure you want to delete 1 account? (%s) Y/y"
```

Comment:
Is "Y/y" at the end of the prompt accurate (shouldn't it be "Y/n", or how user should understand how to opt-out)?
NB: This is not introduced by this PR, just a random bit, that can be fixed (if needed) w/i the scope of the PR's delta.